src/main/java/org/sonarlint/intellij/ui/tree/FlowsTree.java

```diff
@@ -106,22 +108,26 @@ private void navigateToEditor(@Nullable DefaultMutableTreeNode node) {
     if (node == null) {
       return;
     }
-    RangeMarker rangeMarker = null;
+    RangeMarker rangeMarker;
     if (node instanceof FlowNode) {
       var flowNode = (FlowNode) node;
       rangeMarker = flowNode.getFlow().getLocations().stream().findFirst().map(Location::getRange).orElse(null);
     } else if (node instanceof PrimaryLocationNode) {
       var locationNode = (PrimaryLocationNode) node;
       rangeMarker = locationNode.rangeMarker();
+    } else {
+      rangeMarker = null;
     }
     if (rangeMarker == null || !rangeMarker.isValid()) {
       return;
     }
 
-    var psiFile = PsiDocumentManager.getInstance(project).getPsiFile(rangeMarker.getDocument());
-    if (psiFile != null && psiFile.isValid()) {
-      new OpenFileDescriptor(project, psiFile.getVirtualFile(), rangeMarker.getStartOffset()).navigate(false);
-    }
+    runReadActionSafely(project, () -> {
+      var psiFile = PsiDocumentManager.getInstance(project).getPsiFile(rangeMarker.getDocument());
+      if (psiFile != null && psiFile.isValid()) {
+        new OpenFileDescriptor(project, psiFile.getVirtualFile(), rangeMarker.getStartOffset()).navigate(false);
+      }
```

Comment:
This code is duplicated below, maybe we could have a utility method like `openFileFromRangeMarker`?